src/api/test/functional/source_controller_test.rb

```diff
@@ -1140,6 +1140,20 @@ def test_delete_project_with_local_devel_packages
     assert_response :success
     put '/source/home:tom:project/C/_meta', "<package name='C' project='home:tom:project'> <title/> <description/> <devel package='B'/> </package>"
     assert_response :success
+    # delete a package which is used as devel package
+    get '/source/home:tom:project/B/_meta'
+    assert_response :success
+    assert_xml_tag tag: "devel"
+    delete '/source/home:tom:project/A'
+    assert_response 400
+    assert_select "status", code: "delete_error" do
+      assert_select "summary", "Package is used by following packages as devel package: home:tom:project/B"
+    end
+    delete '/source/home:tom:project/A?force=1'
+    assert_response :success
+    get '/source/home:tom:project/B/_meta'
+    assert_response :success
+    assert_no_xml_tag tag: "devel"
```

Comment:
I am looking for ways to make tests more granular. This one isn't actually that big. Still it would be nice to get to smaller, more readable tests.

I am still in shock from looking through channel_maintenance_test...